#!/usr/bin/env python
PACKAGE = "fw_ctrl"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("en_min_ground_speed", bool_t, 0, "enable additional (beyond track keeping) minimum desired forward, on-track ground speed", False)
gen.add("min_ground_speed_g", double_t, 0, "(if enabled) minimum forward, on-track ground speed [m/s]", 0.0, 0.0, 10.0)
gen.add("min_ground_speed_e_max", double_t, 0, "(always enabled) minimum forward, on-track ground speed applied at maximum normalized track error [m/s]", 4.0, 0.0, 10.0)
gen.add("nte_fraction", double_t, 0, "fraction of normalized track error at which min_ground_speed_e_max is applied", 0.5, 0.1, 1.0)
gen.add("lat_time_const", double_t, 0, "lateral-directional guidance time constant [s]", 7.0, 0.1, 20.0)
gen.add("lat_p_gain", double_t, 0, "lateral-directional guidance proportional gain", 0.1, 0.01, 1.0)
gen.add("wind_ratio_buf", double_t, 0, "fraction of wind ratio used as buffer zone for command continuity", 0.1, 0.05, 0.2)

gen.add("lon_time_const", double_t, 0, "longitudinal guidance time constant [s] (not used when vert pos err bnd is fixed)", 5.0, 0.1, 10.0)
gen.add("fpa_app_max", double_t, 0, "maximum approach flight path angle [deg]", 15.0, 1.0, 70.0)
gen.add("fix_vert_pos_err_bnd", bool_t, 0, "use fixed vertical position error boundary", True)
gen.add("vert_pos_err_bnd", double_t, 0, "longitudinal error boundary [m]", 4.0, 1.0, 20.0)

gen.add("en_off_track_roll_ff", bool_t, 0, "enable off-track feed-forward roll commands", True)
gen.add("cross_err_thres", double_t, 0, "track convergence criteria: threshold for cross product of heading error", 0.1, 0.01, 0.5)
gen.add("track_err_thres", double_t, 0, "track convergence criteria: threshold for track error", 0.1, 0.01, 5.0)

exit(gen.generate(PACKAGE, "fw_ctrl", "trajectory_generation"))
